packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -482,6 +485,30 @@ class CameraController extends ValueNotifier<CameraValue> {
     }
   }
 
+  /// TODO: Documentation
+  void onCameraTimeLimitReachedEvent({onCameraTimeLimitReached}) {
+    if (!value.isInitialized || _isDisposed) {
+      throw CameraException(
+        'Uninitialized CameraController',
+        'cameraTimeLimitReachedEventStream was called on uninitialized CameraController',
+      );
+    }
+    if (!value.isRecordingVideo) {
+      throw CameraException(
+        'No video is recording',
+        'cameraTimeLimitReachedEventStream was called when no video is recording.',
+      );
+    }
+
+    CameraPlatform.instance
+        .onCameraTimeLimitReached(_cameraId)
+        .first
+        .then((event) {
+      value = value.copyWith(isRecordingVideo: false);
+      onCameraTimeLimitReached(event.path);
+    });
+  }
```

Comment:
One issue I see with this, is that if the user sets a `maxVideoDuration`, but never calls `onCameraTimeLimitReachedEvent` to handle the callback, `value` will never be updated, causing `isRecordingVideo to still report `true`, while it has in fact already stopped. 

Ideally, the camera controller should always be listening for `CameraPlatform.instance.onCameraTimeLimitReached` to update the CameraValue, even when the user has not provided a callback, so that the state is always correct. 

Additionally, I think I would personally prefer seeing a method that returns a stream of these events (Like how the platform interface exposes events) rather than a method that allows for setting a callback, as that would imo offer a bit more flexibility. What do you think?

Reply:
I agree, this would be easier for users to manage if we can use this as a `Stream<XFile>` and simply listen for a file to arrive. 

We can even extend the `stopVideoRecording` method to add the final video file to the same stream. This way developers simply listen to the stream to get informed that a new video file is available (no matter if the end-user pressed the stop button or if the max time limit has been reached). Of course this would mean the name of the event and the stream should be updated a to reflect this change. Something like:

```dart
Stream<VideoRecordedEvent> onVideoRecorded(int cameraId); 
```

Reply:
This might also solve possible issues with recording video's and stopping them for unusual reasons, like full storage